### PR TITLE
[Datagrid] Allow only int on setPage & setMaxPerPage

### PR DESCRIPTION
**Pull request by tambait**
## Subject

This is a follow up of #6404 to add param & return type hints.

## Changelog

```markdown
### Added
- Added parameter type hints to `PagerInterface` and `Pager`.

### CODE REVIEW

File: tests/Datagrid/DatagridTest.php [hunk 1]
```diff
@@ -502,11 +502,7 @@ public function getBuildPagerWithPageTests(): array
         // tests for php 5.3, because isset functionality was changed since php 5.4
```
Comment (VincentLanglet) [inline tests/Datagrid/DatagridTest.php:502]: This comment should be removed.

File: tests/Datagrid/DatagridTest.php [hunk 2]
```diff
@@ -544,9 +540,6 @@ public function getBuildPagerWithPage2Tests(): array
...
```
Comment (VincentLanglet) [inline tests/Datagrid/DatagridTest.php:540]: Same this could be removed

**Review by VincentLanglet — APPROVED** [review]

**Comment by tambait** [comment]
Not sure I understand why this is failing now. It seems that this one is causing the issue

`The job was canceled because "_7_4_highest_false_sonata-project_block-bundle__4___" failed.`

**Comment by VincentLanglet** [comment]
You can retrigger the build with a close/open. Maybe it will be enough.

**Review by VincentLanglet — APPROVED** [review]

**Review by core23 — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks @tambait

**Pull request merged**